src/data/nat/basic.lean

```diff
@@ -467,166 +468,137 @@ end
 @[simp] lemma pred_one_add (n : ℕ) : pred (1 + n) = n :=
 by rw [add_comm, add_one, pred_succ]
 
-/-! ### `sub` -/
```

Comment:
What is the motivation for deleting some lemmas and keeping some with a new proof?

Reply:
The goal is to remove (basically) all of them, and only use the lemmas that are proven for general types. But I'll do that in future PRs (that is what I meant by "Remove the lemmas specific to each individual type").
I currently replace the proof to show that we have the general lemma.

I have already removed a couple of lemmas, but I'll do the rest in future PRs